### ISSUE TITLE
Suggestion for package title adds a `__test` instead of `_test` only

### ISSUE THREAD

**Issue by dlsniper** [go-lang-plugin-org/go-lang-idea-plugin#2456]
- Plugin version (or commit hash): 0.11.1288
IntelliJ IDEA (Minerva) IU-145.844.1
Build #IU-145.844, built on April 5, 2016
JRE: 1.8.0_76-release-b18 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

- What are you trying to do?
Have a package named `package_test`
- What would you expect to happen?
The only suggestion to be `package_test`
- What happens?
I have two suggestions presented: `package__test` and `package_test` (first one is wrong).

![snapshot25](https://cloud.githubusercontent.com/assets/607868/14459115/fdfd3c5a-00ad-11e6-900f-368996b3ef6f.png)

**Comment by zolotov** [comment]
So what's the bug exactly? During completion you have two packages in current directory: `intent` and `intent_`. Both package names are valid, both packages can be completed by `_test` suffix

**Comment by dlsniper** [comment]
In the current directory (package) I only have a single package, called `intent` and I would like to have the `intent_test` package, since I'm also in a `_test.go` file. And if I'd have a `intent__test` package defined, then the other inspection, `Multiple packages in directory` would kick in as well.

**Comment by zolotov** [comment]
> In the current directory (package) I only have a single package

No, on the screenshot you can see that you have a file with `intent_` package name.

>  And if I'd have a intent__test package defined, then the other inspection, Multiple packages in directory would kick in as well.

And it will as soon as you finish the completion

**Comment by dlsniper** [comment]
> No, on the screenshot you can see that you have a file with `intent_` package name.

I've tried this again. After `package intent` I've typed an ` _ `. The statement became `package intent_` and the completion window suggested the following: `intent_test` and `intent__test`. I'll get a video tonight if it's still not clear.

**Comment by zolotov** [comment]
It's completely clear what happens. It's not clear why do you think it shouldn't happen. What's the reasons not to show intent__test if you got intent_ package as soon as you type _?

Alexander Zolotov

On Вт, 12 апр. 2016 г. at 15:19 "Florin Pățan"

<
mailto:
> wrote:

No, on the screenshot you can see that you have a file with

intent_

package name.

I've tried this again. After

package intent

I've typed an

_

. The statement became

package intent_

and the completion window suggested the following:

intent_test

and

intent__test

. I'll get a video tonight if it's still not clear.

**Comment by dlsniper** [comment]
>  It's not clear why do you think it shouldn't happen.

The context why this is unexpected is:
- current package is named `intent`
- I've just created the file `text_test` (the name I've filled in for the file name prompt when choosing `Go File` from the `New` menu)
- I've set my cursor at the end of the package name so that I can name it `intent_test`
- I've typed `_` after the package name and pressed `tab` in hopes it will be automatically completed to `_test` (since it would have been the only name that should appear, at least to my expectation).

Now, when doing those actions I wasn't expecting to have a suggestion for `intent__test`, just `intent_test`. Like I've said, if I'd choose that item, `intent__test` then the other issue would appear, I would now have two, valid, package different in the same directory.

**Comment by zolotov** [comment]
> I've typed _ after the package name

that's all, now you have `intent_` package. I don't know how to explain it in different way. So let's assign this issue on you and you'll try to implement something to make this completion better. See com.goide.completion.GoCompletionContributor#fillCompletionVariants

**Comment by dlsniper** [comment]
Ok, thanks, I'll try and see if I van work something out.

**Comment by dlsniper** [comment]
I've had a look at this and, to confirm from what I can understand, the issue is that as soon as hit `_` IntelliJ will see the current package as being a new one.
Would it be feasible/desirable/possible to do a check to see if the current file is the only one in the package?

**Comment by zolotov** [comment]
> Would it be feasible/desirable/possible to do a check to see if the current file is the only one in the package?

I think it would be more convenient if you'd try to implement it and write with a test. At the moment I'm not sure I understand your suggestion correctly
